TCPEchoServer/Socket.cpp

```diff
@@ -0,0 +1,95 @@
+#include "pch.h"
+#define WIN32_LEAN_AND_MEAN
+#define _WINSOCK_DEPRECATED_NO_WARNINGS
+#include <ws2tcpip.h>
+#include "Socket.h"
+
+Socket::Socket()
+{
+	m_s = socket(AF_INET, SOCK_STREAM, IPPROTO_TCP);
+}
+
+Socket::Socket(Socket&& other) noexcept
+{
+	m_s = other.m_s;
+	other.m_s = INVALID_SOCKET;
+};
+
+Socket::Socket(SOCKET s)
+{
+	m_s = s;
+}
+
+int Socket::connect(const std::string& ip, u_short port)
+{
+	sockaddr_in hints;
+
+	hints.sin_family = AF_INET;
+	hints.sin_addr.s_addr = inet_addr(ip.c_str());
+	hints.sin_port = htons(port);
+
+	int iResult = ::connect(m_s, (SOCKADDR*)&hints, sizeof(hints));
+	if (iResult == SOCKET_ERROR) {
+		closesocket(m_s);
```

Comment:
Please don't touch a socket here. The only destructor should close the one.

Reply:
To make your code more consistent I would return whatever ::connect return. The user will check the retcode and will call WSAGetLastError if there's a need.

Reply:
Please apply the above comment to the rest of the code.

Reply:
Fixed